npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser/lockfile_parser.rb

```diff
@@ -67,6 +68,8 @@ def lockfile_for(file)
           @lockfiles ||= {}
           @lockfiles[file.name] ||= if [*package_locks, *shrinkwraps].include?(file)
                                       JsonLock.new(file)
+                                    elsif pnpm_locks.include?(file)
+                                      PnpmLock.new(file)
                                     else
                                       YarnLock.new(file)
```

Comment:
What about when these cases conflict such that a user has a lock file for both yarn and pnpm... we don't actually know what they prefer us to do, so should we throw an error in that case?

This is probably true other places in this code as well, and while the issue existed previously, it'll be even more true now that there's three options and `pnpm` being such a close cousin of `npm`... if multiple lockfiles are present, we can't definitively say "npm should win, then pnpm, then yarn"...

Reply:
Definitely.

As you mentioned, there are quite a few places were we establish a package manager priority, unfortunately.

In my opinion, it'd be probably be best to infer a package manager in the file fetcher, and error out there if there's a conflict. And then stop looking at different possibilities everywhere else and then just follow the chosen package manager logic through all the update assuming only the proper files for that package manager are present.

But that'd be probably a separate refactoring task?

Reply:
Sounds reasonable on all counts.

Reply:
I did change the priority so that it's at least consistent for now in the different places.